src/views/connectionUI.ts

```diff
@@ -488,6 +482,22 @@ export class ConnectionUI {
 		});
 	}
 
+	public async addFirewallRule(uri: string, profile: IConnectionProfile): Promise<boolean> {
+		const self = this;
```

Comment:
This shouldn't be necessary (especially since you're using `this` in the code below already anyways...)